src/driver/src/DefaultDriver.cpp

```diff
@@ -55,6 +66,103 @@ YAML::Node loadConfig(const std::string& source,
     }
 }
 
+YAML::Node parseExternal(const YAML::Node& external,
```

Comment:
would it make things any easier to test or re-use if this code were in `ConfigNode` i.e. if you could just do `ConfigNode parsed = ConfigNode::load(filePath)`?